In [3]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
pd.set_option('display.max_columns', None)

df = pd.read_csv('diabetes_cleaned_numeric_bin.csv')

In [33]:
df.head()

gender  age  time_in_hospital  num_lab_procedures  num_procedures  \
0       0   80                13                  68               2   
1       0   90                12                  33               3   
2       1   40                 1                  51               0   
3       0   40                 9                  47               2   
4       1   50                 3                  31               6   

   num_medications  number_outpatient  number_emergency  number_inpatient  \
0               28                  0                 0                 0   
1               18                  0                 0                 0   
2                8                  0                 0                 0   
3               17                  0                 0                 0   
4               16                  0                 0                 0   

   number_diagnoses  nateglinide  chlorpropamide  acarbose  \
0                 8            0               0         0   
1                 8            0               0         0   
2                 5            0               0         0   
3                 9            0               0         0   
4                 9            0               0         0   

   glyburide-metformin  change  diabetesMed  readmitted  race_?  \
0                    0       1            1           0       0   
1                    0       1            1           0       0   
2                    0       1            1           0       0   
3                    0       0            1           0       0   
4                    0       0            1           0       0   

   race_AfricanAmerican  race_Asian  race_Hispanic  race_Other  \
0                     0           0              0           0   
1                     0           0              0           0   
2                     0           0              0           0   
3                     1           0              0           0   
4                     0           0              0           0   

   admission_type_id_2  admission_type_id_3  admission_type_id_4  \
0                    1                    0                    0   
1                    0                    1                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    1                    0                    0   

   admission_type_id_5  admission_type_id_6  admission_type_id_7  \
0                    0                    0                    0   
1                    0                    0                    0   
2                    0                    0                    0   
3                    0                    0                    0   
4                    0                    0                    0   

   admission_type_id_8  discharge_disposition_id_2  \
0                    0                           0   
1                    0                           0   
2                    0                           0   
3                    0                           0   
4                    0                           0   

   discharge_disposition_id_3  discharge_disposition_id_4  \
0                           0                           0   
1                           1                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

   discharge_disposition_id_5  discharge_disposition_id_6  \
0                           0                           0   
1                           0                           0   
2                           0                           0   
3                           0                           0   
4                           0                           0   

   discharge_disposition_id_7  discharge_disposition_id_8

In [4]:
df.shape

(69970, 149)

In [34]:
df.readmitted.value_counts()

0    63693
1     6277
Name: readmitted, dtype: int64

In [35]:
from sklearn.model_selection import train_test_split

X = df.drop(columns='readmitted')
y = df.readmitted
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=42)

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix


estimator = LogisticRegression(C=1e5,class_weight='balanced',multi_class='ovr',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=5,verbose=1)
selector = selector.fit(X_train, y_train)

In [ ]:
from sklearn.metrics import fbeta_score, make_scorer
ftwo_scorer = make_scorer(fbeta_score, beta=2)
from sklearn.model_selection import GridSearchCV
from sklearn.svm import LinearSVC
grid = GridSearchCV(LinearSVC(), param_grid={'C': [1, 10]},scoring=ftwo_scorer, cv=5)

In [42]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=1e5,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=5,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 69 features.
Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 fe

Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
F

Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 features.
Fitting estimator with 40 features.
Fitting estimator with 39 features.
Fitting estimator with 38 features.
Fitting estimator with 37 features.
Fitting estimator with 36 features.
Fitting estimator with 35 features.
Fitting estimator with 34 features.
Fitting estimator with 33 features.
Fitting estimator with 32 features.
Fitting estimator with 31 features.
Fitting estimator with 30 fe

In [58]:
best = selector.estimator_
print(best.coef_)
best

[[ 0.02020315  0.02077738 -0.00980075  0.02105735  0.13470875  0.34740949
   0.05423375 -0.13809256 -1.03696109 -0.35625888 -0.24711554 -0.04574416
   0.32775655 -0.17126646 -0.13897639 -0.22308456 -0.04322864 -0.14558424
  -6.71628087 -0.22047037  0.20555604 -8.3879774  -0.06381435  0.67134057
   0.69426951  0.45421394  1.20503673  0.21469271  0.25443127  0.39160825
   2.17015576 -7.46051487 -5.01005066  2.23637825 -5.85077599 -7.18814775
   0.54091534  1.63328117 -1.01418603 -7.77490383 -0.35857364 -5.66135727
   1.83212164  0.10385818 -0.06922132  0.63420185 -0.10048276 -0.39965994
   0.09288383  0.07306536  0.48255927 -6.79257531 -3.23276456 -4.97786776
  -0.20529774  0.39063202 -6.49611763 -5.61505015  0.36851676  0.19720245
   0.25873844 -0.080407   -0.10306213 -0.0171654   0.13203156 -0.11949255
  -0.21929032 -0.88995244  0.27731741  0.71274144  0.23894045 -0.08022993
   0.2276632   0.16943271  0.03396107  0.02904667 -0.13915833  0.04781743
  -1.403413   -0.13370494  0.08389899 

LogisticRegression(C=100000.0, class_weight='balanced', dual=False,
                   fit_intercept=True, intercept_scaling=1, l1_ratio=None,
                   max_iter=100, multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=5, solver='liblinear', tol=0.0001, verbose=0,
                   warm_start=False)

In [48]:
selector.ranking_

array([ 1,  4,  1,  8,  1,  6,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        2,  7,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  1,  1,  1,  1,  1,  1, 10,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  9,  1,
        1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  3,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,  1,
        1,  1,  1,  1,  1,  1,  1,  5,  1,  1,  1,  1])

In [49]:
result = pd.Series(selector.ranking_,index=X_train.columns).sort_values()
result

gender                     1
insulin_Up                 1
diag_1_?                   1
diag_1_Blood               1
diag_1_Congenital          1
                          ..
num_medications            6
race_Asian                 7
num_lab_procedures         8
glyburide_Steady           9
admission_source_id_13    10
Length: 148, dtype: int64

In [59]:
X_train.columns[selector.support_]

Index(['gender', 'time_in_hospital', 'num_procedures', 'number_outpatient',
       'number_emergency', 'number_inpatient', 'number_diagnoses',
       'nateglinide', 'chlorpropamide', 'acarbose',
       ...
       'diag_3_Genitourinary', 'diag_3_Infectious', 'diag_3_Injury_Poison',
       'diag_3_Mental_Disorders', 'diag_3_Muscles_Skeleton',
       'diag_3_Neoplasms', 'diag_3_Pregnancy', 'diag_3_Respiratory',
       'diag_3_Skin', 'diag_3_Supplementary'],
      dtype='object', length=139)

In [52]:
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

array([[34791, 16177],
       [ 2282,  2726]])

In [57]:
selector.classes_

array([0, 1])

In [60]:
selector.grid_scores_

array([0.49985557, 0.4999831 , 0.50004026, 0.50008931, 0.50013836,
       0.5001776 , 0.50020703, 0.50021684, 0.50025608, 0.50028551,
       0.50034437, 0.50034437, 0.50035418, 0.50082454, 0.5010425 ,
       0.50150735, 0.50241953, 0.50600645, 0.5148351 , 0.52011116,
       0.52478169, 0.52925973, 0.53378199, 0.53378199, 0.53378199,
       0.53622549, 0.5453752 , 0.55161543, 0.55324941, 0.5731806 ,
       0.57515241, 0.58109991, 0.58197305, 0.58152483, 0.59048898,
       0.59013582, 0.59209401, 0.5932692 , 0.59636408, 0.60030485,
       0.59969363, 0.60014488, 0.60043728, 0.6004995 , 0.60006075,
       0.60003395, 0.6007717 , 0.60055026, 0.60136255, 0.60252929,
       0.60297012, 0.60257842, 0.60216381, 0.60218352, 0.60196644,
       0.60182327, 0.6015349 , 0.60096892, 0.60055944, 0.59974423,
       0.60048171, 0.60009182, 0.60023928, 0.59978985, 0.5992627 ,
       0.59937821, 0.59974785, 0.60051622, 0.60039118, 0.59988878,
       0.60028275, 0.59965402, 0.59907003, 0.59877375, 0.60004

In [63]:
selector.decision_function(X_train)

array([-0.70433585,  1.46514278,  0.14263829, ..., -0.36358608,
       -0.04713022, -0.80302059])

### Round 2

In [ ]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.feature_selection import RFE
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=1e5,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFE(estimator,n_features_to+select=50,step=1)estimator, step=1, cv=5,verbose=1,scoring=roc_scorer
selector = RFECV(estimator, step=1, cv=5,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

In [ ]:
>>> from sklearn.datasets import make_friedman1
>>> from sklearn.feature_selection import RFE
>>> from sklearn.svm import SVR
>>> X, y = make_friedman1(n_samples=50, n_features=10, random_state=0)
>>> estimator = SVR(kernel="linear")
>>> selector = RFE(estimator, n_features_to_select=5, step=1)
>>> selector = selector.fit(X, y)
>>> selector.support_
array([ True,  True,  True,  True,  True, False, False, False, False,
       False])
>>> selector.ranking_
array([1, 1, 1, 1, 1, 6, 4, 3, 2, 5])

In [64]:
from sklearn.linear_model import LogisticRegression
from sklearn.model_selection import GridSearchCV
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

params = {'C':[0.001,0.005,0.01,0.05,0.1,0.5,1,5,10,50]}
estimator = LogisticRegression(class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1,cv=3,verbose=1,scoring=roc_scorer)
clf = GridSearchCV(selector, param_grid=params, scoring=roc_scorer,cv=3,verbose=1)
clf.fit(X_train,y_train)


Fitting 3 folds for each of 10 candidates, totalling 30 fits


[Parallel(n_jobs=1)]: Using backend SequentialBackend with 1 concurrent workers.


ValueError: Invalid parameter C for estimator RFECV(cv=3,
      estimator=LogisticRegression(C=1.0, class_weight='balanced', dual=False,
                                   fit_intercept=True, intercept_scaling=1,
                                   l1_ratio=None, max_iter=100,
                                   multi_class='warn', n_jobs=None,
                                   penalty='l2', random_state=5,
                                   solver='liblinear', tol=0.0001, verbose=0,
                                   warm_start=False),
      min_features_to_select=1, n_jobs=None, scoring=make_scorer(roc_auc_score),
      step=1, verbose=1). Check the list of available parameters with `estimator.get_params().keys()`.

In [ ]:
grid = {"estimator__max_depth": [3, None],
        "estimator__min_samples_split": sp_randint(1, 11),
        "estimator__min_samples_leaf": sp_randint(1, 11),
        "estimator__bootstrap": [True, False],
        "estimator__criterion": ["gini", "entropy"]}

estimator = RandomForestClassifier()
selector = RFECV(estimator, step=1, cv=4)
clf = RandomizedSearchCV(selector, param_distributions=grid, cv=7)

In [65]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.001,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 fe

Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
Fitting estimator with 109 features.
F

In [66]:
selector.grid_scores_

array([0.53659044, 0.56504525, 0.58788183, 0.58938544, 0.59290577,
       0.59156682, 0.59391681, 0.59268755, 0.59513987, 0.59527528,
       0.59407819, 0.59329623, 0.59381241, 0.59382046, 0.59302935,
       0.59282346, 0.59316154, 0.5932718 , 0.59313578, 0.59341979,
       0.59333531, 0.59426613, 0.59371417, 0.59300375, 0.59321816,
       0.59400175, 0.59360714, 0.59388743, 0.59435966, 0.59530051,
       0.59556074, 0.59551901, 0.59348484, 0.59235259, 0.59295725,
       0.59282161, 0.59315691, 0.59295132, 0.59300213, 0.59300219,
       0.59271254, 0.59218387, 0.59230329, 0.59157314, 0.59240145,
       0.59246195, 0.59192926, 0.59183892, 0.59198478, 0.59178676,
       0.59224482, 0.59157213, 0.59197333, 0.59171487, 0.591777  ,
       0.59244278, 0.59225463, 0.59226596, 0.59227577, 0.59242292,
       0.5925263 , 0.59317433, 0.5931466 , 0.59306648, 0.59287663,
       0.5922172 , 0.59286718, 0.59232863, 0.59184737, 0.59184738,
       0.59224311, 0.5924937 , 0.59265242, 0.59277644, 0.59338

In [67]:
best = selector.estimator_
print(best.coef_)
best

[[ 0.0876437   0.31760124  0.13944601 -0.06050937 -0.0683795   0.14530402
   0.34443577  0.19774276  0.09789688  0.43359493  0.04986591 -0.05238731
  -0.08475984  0.05207432 -0.05502914  0.07525487 -0.0690985  -0.20265631
  -0.05240117 -0.05379664 -0.09770693 -0.05806362 -0.05089316  0.0946097
  -0.07177012  0.06971677 -0.08240005 -0.06689188 -0.04830608  0.06966018
  -0.08305235]]


LogisticRegression(C=0.001, class_weight='balanced', dual=False,
                   fit_intercept=True, intercept_scaling=1, l1_ratio=None,
                   max_iter=100, multi_class='warn', n_jobs=None, penalty='l2',
                   random_state=5, solver='liblinear', tol=0.0001, verbose=0,
                   warm_start=False)

In [68]:
result = pd.Series(selector.ranking_,index=X_train.columns).sort_values()
result

diag_3_Supplementary             1
discharge_disposition_id_2       1
discharge_disposition_id_3       1
discharge_disposition_id_5       1
diag_1_Skin                      1
                              ... 
admission_source_id_14         114
admission_source_id_11         115
discharge_disposition_id_12    116
num_lab_procedures             117
admission_source_id_13         118
Length: 148, dtype: int64

In [69]:
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

array([[36429, 14539],
       [ 2573,  2435]])

In [70]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.01,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 fe

Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
Fitting estimator with 109 features.
F

In [71]:
selector.grid_scores_

array([0.52168024, 0.53010736, 0.56531391, 0.57743568, 0.59631796,
       0.59661402, 0.59725199, 0.59849933, 0.59883274, 0.59901867,
       0.59915584, 0.59967129, 0.59974149, 0.60062865, 0.59995516,
       0.59874776, 0.59836426, 0.59785331, 0.59864248, 0.59808726,
       0.59810842, 0.59872581, 0.59857356, 0.59695844, 0.59728217,
       0.59668681, 0.59688126, 0.5963121 , 0.5990497 , 0.60035985,
       0.60069309, 0.60069954, 0.60042344, 0.60095121, 0.60129043,
       0.60048216, 0.60048596, 0.60091855, 0.60016555, 0.60015194,
       0.6002225 , 0.59994322, 0.59937179, 0.59919169, 0.59963803,
       0.59975223, 0.59994827, 0.59993402, 0.59973148, 0.5989356 ,
       0.59894365, 0.59924811, 0.59901726, 0.59852777, 0.59856026,
       0.59796173, 0.59916803, 0.60165594, 0.60071371, 0.60074944,
       0.6013896 , 0.60206373, 0.60259409, 0.60242217, 0.60244003,
       0.6016851 , 0.60188005, 0.60213251, 0.60299831, 0.60350918,
       0.6025109 , 0.60266146, 0.60256494, 0.60282176, 0.60176

In [72]:
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

array([[34894, 16074],
       [ 2311,  2697]])

In [73]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.1,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 fe

In [74]:
selector.grid_scores_

array([0.52168024, 0.5221454 , 0.5248491 , 0.52639111, 0.53311488,
       0.54385258, 0.5491862 , 0.56392718, 0.57513319, 0.57515792,
       0.5757839 , 0.5756811 , 0.57583353, 0.57560162, 0.58133954,
       0.58147484, 0.58991376, 0.59751349, 0.59780825, 0.59840853,
       0.5993264 , 0.5986741 , 0.59811346, 0.59807524, 0.59811408,
       0.59774867, 0.5987403 , 0.59876343, 0.59960051, 0.59932544,
       0.60032757, 0.59992801, 0.60070096, 0.6006172 , 0.60090955,
       0.60087287, 0.60061636, 0.60048178, 0.6008857 , 0.60115682,
       0.60019948, 0.60043153, 0.59913618, 0.6000041 , 0.60011188,
       0.60020965, 0.5999693 , 0.60033743, 0.60002453, 0.59974934,
       0.60019403, 0.60064875, 0.60136464, 0.60097809, 0.60110763,
       0.60058467, 0.59976165, 0.59976107, 0.6003765 , 0.60072757,
       0.60155407, 0.60090586, 0.60141951, 0.60124116, 0.60063593,
       0.60082408, 0.60129848, 0.59993412, 0.60080261, 0.60048666,
       0.60072772, 0.60020246, 0.60095195, 0.60082254, 0.60086

In [75]:
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

array([[34703, 16265],
       [ 2258,  2750]])

In [79]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.0001,class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 fe

Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
Fitting estimator with 109 features.
F

In [80]:
selector.grid_scores_

array([0.54536078, 0.56819736, 0.58788183, 0.58709758, 0.57305701,
       0.56985867, 0.55205198, 0.5586103 , 0.56011604, 0.56169083,
       0.56303375, 0.56393288, 0.56418157, 0.56559762, 0.56689716,
       0.56768911, 0.56740354, 0.56866797, 0.56917765, 0.5682323 ,
       0.56757825, 0.56778703, 0.56987199, 0.57096506, 0.57103014,
       0.57091707, 0.57040996, 0.57078256, 0.5714958 , 0.57046325,
       0.5706924 , 0.57086722, 0.57060688, 0.57071473, 0.57069341,
       0.57116159, 0.57108952, 0.57057865, 0.57104488, 0.57096294,
       0.57154852, 0.57172039, 0.57102003, 0.57101827, 0.5708676 ,
       0.57093639, 0.57068586, 0.57068762, 0.57070706, 0.57077573,
       0.57069374, 0.57088364, 0.57068223, 0.57063318, 0.57071166,
       0.57061191, 0.57081969, 0.57083749, 0.57093735, 0.57097659,
       0.57097489, 0.57101413, 0.57075555, 0.57079479, 0.57088309,
       0.57081266, 0.57216046, 0.5721016 , 0.57211141, 0.57217027,
       0.5721997 , 0.57217027, 0.57258909, 0.57270857, 0.57292

In [81]:
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

array([[39358, 11610],
       [ 2987,  2021]])

0.0001 (4 features)
array([[39358, 11610],
       [ 2987,  2021]])
0.001 (32 features)
array([[36429, 14539],
       [ 2573,  2435]])
0.01 (101 features)
array([[34894, 16074],
       [ 2311,  2697]])
0.1 (148 features)
array([[34703, 16265],
       [ 2258,  2750]])
       
L1 0.1 (96 features)
array([[35185, 15783],
       [ 2329,  2679]])

In [82]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.0001, penalty='l1',class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 features.
Fitting estimator with 41 fe

Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
Fitting estimator with 109 features.
Fitting estimator with 108 features.
F

In [83]:
print(selector.grid_scores_)
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

[0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255
 0.50029255 0.50029255 0.50029255 0.50029255 0.50029255 0.5002

array([[    0, 50968],
       [    0,  5008]])

In [84]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.001, penalty='l1',class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 69 features.
Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 fe

Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
F

In [85]:
print(selector.grid_scores_)
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

[0.54536078 0.5        0.54428353 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139
 0.54783139 0.54783139 0.54783139 0.54783139 0.54783139 0.5478

array([[28301, 22667],
       [ 2307,  2701]])

In [86]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.01, penalty='l1',class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 69 features.
Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 fe

Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.


In [87]:
print(selector.grid_scores_)
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

[0.52168024 0.55688679 0.57132696 0.59373992 0.59342227 0.59441472
 0.59477666 0.59461768 0.59497537 0.59520618 0.59464459 0.59474677
 0.59518504 0.59503202 0.59461498 0.59461549 0.59452896 0.59460949
 0.59558999 0.59546673 0.59552735 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.5956994  0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926
 0.59579926 0.59579926 0.59579926 0.59579926 0.59579926 0.5957

array([[35765, 15203],
       [ 2484,  2524]])

In [88]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=0.1, penalty='l1',class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 69 features.
Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 fe

Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
Fitting estimator with 121 features.
Fitting estimator with 120 features.
Fitting estimator with 119 features.
Fitting estimator with 118 features.
Fitting estimator with 117 features.
Fitting estimator with 116 features.
Fitting estimator with 115 features.
Fitting estimator with 114 features.
Fitting estimator with 113 features.
Fitting estimator with 112 features.
Fitting estimator with 111 features.
Fitting estimator with 110 features.
F

In [89]:
print(selector.grid_scores_)
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

[0.51492254 0.5221454  0.5248491  0.52922512 0.53311488 0.54777901
 0.57534091 0.5826769  0.59107794 0.58978323 0.59902104 0.5988675
 0.5986571  0.59997913 0.59972011 0.60043373 0.59970101 0.59919695
 0.60013861 0.59998045 0.60022577 0.60062847 0.60206538 0.60146112
 0.60133033 0.60157663 0.60111141 0.60151311 0.60109101 0.60187228
 0.60134599 0.60161663 0.60199704 0.60197322 0.60316059 0.60260549
 0.60254395 0.60229878 0.60172108 0.60187906 0.60174813 0.60149766
 0.60185468 0.60239082 0.60203421 0.60166756 0.60172901 0.60155185
 0.60122303 0.60129333 0.6018808  0.60230825 0.6027298  0.60292765
 0.6026211  0.60280479 0.60310895 0.60306437 0.60308399 0.60297094
 0.6047674  0.60483783 0.60618674 0.6063226  0.60780224 0.60757303
 0.60807068 0.60774156 0.60807726 0.60881194 0.60907858 0.60857908
 0.60858896 0.60827956 0.60763171 0.60731428 0.60666435 0.60646361
 0.60603486 0.60597577 0.60545521 0.60460629 0.6047845  0.60420532
 0.60462244 0.60683064 0.60672103 0.60656406 0.60684748 0.60668

array([[35185, 15783],
       [ 2329,  2679]])

In [90]:
from sklearn.linear_model import LogisticRegression
from sklearn.feature_selection import RFECV
from sklearn.metrics import confusion_matrix
from sklearn.metrics import make_scorer
from sklearn.metrics import roc_auc_score

roc_scorer = make_scorer(roc_auc_score)

estimator = LogisticRegression(C=1, penalty='l1',class_weight='balanced',solver='liblinear',random_state=5)
selector = RFECV(estimator, step=1, cv=3,verbose=1,scoring=roc_scorer)
selector = selector.fit(X_train, y_train)

Fitting estimator with 148 features.
Fitting estimator with 147 features.
Fitting estimator with 146 features.
Fitting estimator with 145 features.
Fitting estimator with 144 features.
Fitting estimator with 143 features.
Fitting estimator with 142 features.
Fitting estimator with 141 features.
Fitting estimator with 140 features.
Fitting estimator with 139 features.
Fitting estimator with 138 features.
Fitting estimator with 137 features.
Fitting estimator with 136 features.
Fitting estimator with 135 features.
Fitting estimator with 134 features.
Fitting estimator with 133 features.
Fitting estimator with 132 features.
Fitting estimator with 131 features.
Fitting estimator with 130 features.
Fitting estimator with 129 features.
Fitting estimator with 128 features.
Fitting estimator with 127 features.
Fitting estimator with 126 features.
Fitting estimator with 125 features.
Fitting estimator with 124 features.
Fitting estimator with 123 features.
Fitting estimator with 122 features.
F

Fitting estimator with 69 features.
Fitting estimator with 68 features.
Fitting estimator with 67 features.
Fitting estimator with 66 features.
Fitting estimator with 65 features.
Fitting estimator with 64 features.
Fitting estimator with 63 features.
Fitting estimator with 62 features.
Fitting estimator with 61 features.
Fitting estimator with 60 features.
Fitting estimator with 59 features.
Fitting estimator with 58 features.
Fitting estimator with 57 features.
Fitting estimator with 56 features.
Fitting estimator with 55 features.
Fitting estimator with 54 features.
Fitting estimator with 53 features.
Fitting estimator with 52 features.
Fitting estimator with 51 features.
Fitting estimator with 50 features.
Fitting estimator with 49 features.
Fitting estimator with 48 features.
Fitting estimator with 47 features.
Fitting estimator with 46 features.
Fitting estimator with 45 features.
Fitting estimator with 44 features.
Fitting estimator with 43 features.
Fitting estimator with 42 fe

In [91]:
print(selector.grid_scores_)
best = selector.estimator_
predicts = best.predict(X_train[X_train.columns[selector.support_]])
confusion_matrix(y_train,predicts)

[0.50018997 0.5009217  0.50145394 0.50149317 0.50153241 0.50907982
 0.51629419 0.52375784 0.52498105 0.52769456 0.5334376  0.53381095
 0.53387508 0.54461789 0.54987632 0.56377227 0.56453728 0.57570423
 0.57580732 0.57711722 0.57740928 0.5770841  0.58154102 0.58189072
 0.58949668 0.59022171 0.59019082 0.59767    0.59912638 0.59862048
 0.59841763 0.59838394 0.59824247 0.59923465 0.59967084 0.59996668
 0.60006767 0.60087362 0.60131501 0.60130269 0.60095228 0.60087345
 0.60095193 0.60121186 0.60009555 0.59990006 0.59988929 0.6002994
 0.60036977 0.60067887 0.60094791 0.60022699 0.60065985 0.60178314
 0.60149367 0.6007542  0.60059474 0.60064674 0.60090597 0.60064899
 0.60037062 0.60030824 0.6017152  0.60121837 0.60060763 0.60127774
 0.60091288 0.6018904  0.6015203  0.60113879 0.60083122 0.6007152
 0.60107762 0.60043751 0.60073181 0.60131134 0.6010594  0.60133017
 0.60097626 0.60117417 0.60158591 0.60188538 0.60220452 0.6023811
 0.60269014 0.60235337 0.60204756 0.60291482 0.60320335 0.6033073

array([[34748, 16220],
       [ 2262,  2746]])

In [1]:
df.columns

NameError: name 'df' is not defined